lib/shelf-file.js

```diff
@@ -0,0 +1,112 @@
+"use strict";
+
+module.exports = (logger, requestPromise, ShelfMetadata, utils) => {
+    /**
+     * An object that represents a Shelf artifact.
+     */
+    class ShelfFile {
+        /**
+         * @param {string} path The path where this file is located.
+         * @param {string} host The URL where the Shelf instance is located.
+         * @param {string} uri The full URL we're accessing.
+         * @param {string} authToken The token used to authenticate access.
+         */
+        constructor(path, host, uri, authToken) {
+            this.authToken = authToken;
+            this.host = host;
+            this.metadata = new ShelfMetadata(uri, authToken);
+            this.path = path;
+            this.uri = uri;
+        }
+
+
+        /**
+         * Uploads content to shelf at the provided path.
+         *
+         * @param {string|ReadStream} content
+         * @return {Promise.<Object>}
```

Comment:
To describe the `Object` not as an `Object` but to make it return a typedef for instance.  Saying it returns an `Object` is analogous to saying it returns "something" lol.

I think we have since agreed on this though